doc/manual/contract.md

```diff
@@ -0,0 +1,175 @@
+
+To define an invariant we use the annotation `invariant`. Here is an example with our Stack. The class defines that it is impossible to be in a state where the size is less than 0.
+
+~~~
+	class Stack
+		invariant(size >= 0)
+	end
+~~~
+
+Another example with a class that represents a date (day, month). To be in a consistent state, the date must be represented with a day between 1 and 31 and a month between 1 and 12.
+
+~~~
+	class Date
+		invariant(day >= 1 and day <= 31)
+		invariant(month >= 1 and month <= 12)
+	end
+~~~
+
+### Additional informations
+
+* The invariant is only checked at the end of a constructor.
+
+* By default, the invariants are only checked on exit. See section `Verification policy` to activate them on in and out.
+
+## Inheritance
+
+The Nit contracts fully support inheritance (single and multiple) and redefinition. The specialization can be interpreted as like a subcontracting. Subcontracting element must, then, keep the set of promises initially defined by all previous contractors. Here are the subtyping rules for each type of the contracts:
+
+* The `expect` can be weakened, which guarantees that all the contracts introduced in the previous definitions will always be considered as valid entries. During the specialization of a method it will be possible to redefine its precondition in order to open the allowed input.
+
+* The `ensure` can be reinforced, which guarantees that all of the expected results introduced in the previous definitions will be respected. During the specialization of a method it will be possible to redefine its postcondition in order to restrict the possible outputs.
+
+* The `invariant` can be reinforced, which guarantees that constraints of validated state previously defined in the superclasses/interfaces will be respected. The specialization of a class makes it possible to restrict all of the valid states. The objective is that an object must always be considered as a coherent value of all of its superclasses/interfaces.
+
+Nit however offers the possibility of removing the inheritance of contracts with the `no contract`  annotation (use this annotation only when it's really necessary).
+
+## Runtime
+
+When performing the evaluation of a contract, all routines and elements will be executed without the evaluation of contracts. The objective is to remove the potential risk of falling into an infinite loop of verification.
+
+### Execution order
+
+For the public methods/attribute, the evaluation order is as follows:
+
+1- invariant
+2- preconditions (`expect`)
+3- called property
+4- postconditions (`ensure`)
+5- invariant
+
+For constructors only the first step is ignored.
+
+### Verification policy
+
+Since contracts are expensive in resource, Nit offers several levels of contract verification in order to adapt them to the context:
```

Comment:
Enlève la partie "in order to adapt them to the context", ces mots n'appartent pas de valeur à la compréhension.